### DISCUSSION TITLE
nRF52833 support possible or is it too much resource constraints?

### DISCUSSION THREAD

**Discussion by Hedda** [ffenix113/zigbee_home#19] [General]
nRF52833 is another popular nRF5 SoC in the nRF52 series that can be attractive because it costs much less than the nRF52840.

Zigbee is at least supported on nRF52833 in nRF5 SDK:

https://www.nordicsemi.com/Products/Wireless/Zigbee

https://www.nordicsemi.com/Products/Development-software/nrf5-sdk-for-thread-and-zigbee

The problem with little-brother nRF52833 is that is features half as much RAM and Flash compared to big-brother nRF52840.

nRF52833 includes 512 KB Flash + 128 KB RAM, (with otherwise the same 32-bit Ar Cortex-M4 with FPU @ 64-MHz as nRF52840).

E73-2G4M08S1E and E73-2G4M08S1EX radio modules can already be used in "b-parasite" Zigbee plant/flower sensors, see -> https://github.com/ffenix113/zigbee_home/discussions/5

PS: nRF52833 is also features Bluetooth Direction Finding which nRF52840 does not (however the newer nRF5340 does as well):

https://www.nordicsemi.com/Products/Wireless/Bluetooth-Direction-Finding

https://www.seeedstudio.com/blog/2020/09/30/getting-started-with-nrf52833/

**Comment by ffenix113** [comment]
Hello Hedda,

Unfortunately I don't have nRF52833 device on hand to test against(at least at this time), but from what I currently have the firmware generated with 2 sensors adds up to ~260kb flash and ~70kb RAM. 

So if it would be possible to compile the firmware for nrf52833 - I think it has good chances to being supported, though with possibly limited future functionality.

**Comment by Hedda** [comment]
FYI, there is the NRF52 simulated board (BabbleSim) "allow simulating a nRF52833 SOC a Zephyr target boards for development:

https://docs.zephyrproject.org/latest/boards/posix/nrf_bsim/doc/nrf52_bsim.html#nrf52-simulated-board-babblesim

Otherwise check out these relativly inexpensive nRF52833 based boards and radio modules hardware:

BBC Micro:Bit v2 (BBC Micro-bit v2) and later is based on nRF52833, it is sold practically everywhere and it even comes with a battery:

https://microbit.org/new-microbit/

https://microbit.org/buy/

https://microbit.org/

https://www.nordicsemi.com/Nordic-news/2020/10/nordic-powered-next-generation-microbit-to-support-ai-and-machine-learning-applications

https://github.com/Spivoxity/baremetal-v2

There are inexpensive starter/development kits like this for the BBC micro:bit v2 with breadboard adapter or you can buy is standalone

https://www.amazon.com/KEYESTUDIO-Beginner-Starter-Kit-Micro/dp/B07KG2M7BM

https://www.amazon.com/KEYESTUDIO-Starter-Kit-Breadboard-etc-Electronics-Kit-Tutorial/dp/B08H8LK5B5

https://www.amazon.com/Freenove-Ultimate-Starter-Kit-Micro/dp/B081GKJ3WB/

https://www.amazon.com/Freenove-Super-Starter-Kit-Micro/dp/B09YRCF9Y6

https://www.amazon.com/GeeekPi-Basic-Starter-Kit-Micro/dp/B0BKG9L5GY

nRF52833-DK (official nRF52833 development kit) from Nordic Semiconductor

https://www.nordicsemi.com/Products/Development-hardware/nrf52833-dk

https://docs.zephyrproject.org/2.7.5/boards/arm/nrf52833dk_nrf52833/doc/index.html

Tip if you live in the US then someone is selling one cheap on eBay right now:

https://www.ebay.com/itm/195969065810

Raytac have many nRF52833 based radio modules and a few demo/evualuation boards (several of which can find on eBay and Amazon):

https://www.raytac.com/product/

https://docs.zephyrproject.org/latest/boards/arm/raytac_mdbt50q_db_33_nrf52833/doc/index.html

https://www.amazon.com/MDBT50-DB-Nordic-nRF52833-Bluetooth-Antenna/dp/B0B4B2YTCL?th=1

https://www.amazon.com/MDBT50Q-DB-Nordic-nRF52833-Bluetooth-Antenna/dp/B0B49WJP1X/ref=sr_1_1?crid=3BE7TASHLCNX3&keywords=nRF52833&qid=1706879601&sprefix=nrf52833%2Caps%2C347&sr=8-1

So does Fanstel Corp. that can be bought from parts resellers like DigiKey, Fanell, RS-online Arrow, and Mouser Electronics, etc.

https://www.fanstel.com/bluenor-summaries

And u-box too, with u-blox EVK NINA-B40x

https://www.u-blox.com/en/product/usb-anna-b4

https://docs.zephyrproject.org/latest/boards/arm/ubx_evkninab4_nrf52833/doc/index.html

E73-2G4M08S1E and E73-2G4M08S1EX radio modules:

https://www.cdebyte.com/products/E73-2G4M08S1E

https://www.cdebyte.com/products/E73-2G4M08S1EX

https://www.ebyte.com/en/product-view-news.html?id=1035

https://www.ebyte.com/product-view-news.html?id=1034

RF-BM-ND10 and RF-BM-ND07 radio modules from RF-star/RFstar

https://www.szrfstar.com/product/nrf52833_bluetooth_zigbee_thread_aoa_aod_rf_module-en.html

https://www.szrfstar.com/product/316-en.html

I have no idea what bootloader these ship with but maybe the "b-parasite" project has more information on bootloader?

It is very possible that radio modules from Ebyte ship with a custom bootloader and need an SWD programmer to flash it the first time.

E.g. first flashing the Adafruit nRF52 Bootloader with a JLink debug probe -> https://github.com/adafruit/Adafruit_nRF52_Bootloader

**Comment by Hedda** [comment]
As for DIY hardware based on nRF52833, check out these projects:

* https://github.com/spbnick/breadebyte

* https://github.com/rbaron/b-parasite/tree/hardware-v2

* https://github.com/smartboxchannel/eON-AQSENS_213_V2_ED3-EFEKTALAB

* https://github.com/smartboxchannel/eON-INDOOR_CLIMATE_SENSOR_290-EFEKTALAB

* https://github.com/smartboxchannel/EFEKTA-EINK154-TEMP-HUM-PRES-SENSOR-NRF52_V4
